### DISCUSSION TITLE
How to handle Payment Terms?

### DISCUSSION THREAD

**Discussion by andrelam** [kimai/kimai#4663] [Questions & Answers]
Hello, everyone.

There's one thing I really miss in Kimai: the ability to provide a (somewhat) configurable payment term per customer. For example, currently I have 4 specific customers:

Customer A: all invoices have a fixed payment date, which is the 10th of the month subsequent to the invoice date, and if the 10th falls on a weekend, it should be moved to the next Monday. For example, if the invoice date is February 26th, 2024, the due date must be March 11th, 2024 (10 is a Sunday); if the invoice date is March 1st, 2024, the due date must be April 10th, 2024.

Customer B: all invoices have a fixed payment date, which is the 16th of the same month if the invoice is issued before the 10th of the month. If the invoice is issued on or after the 10th of the month, the due date will be the 16th of the next month. If the due date falls on a weekend, the due date should be brought back to the preceding Friday (if the due date is March 16th, it should be moved to March 15th).

Customer C: all invoices have a variable due date, which is 10 days after the invoice date. If the due date falls on a weekend, it should be moved to the next Monday.

Customer D: all invoices have the last Wednesday of the month as due date, if the invoice was issued before the 15th of the month. Otherwise, it will be the last Wednesday of the next month.

Unfortunately, these are real examples, and whenever I create an invoice for these customers I need to remember the specific rule for the customer, check on the calendar what would be the due date for the invoice and inform it during the invoice creation. I handle if manually, but since I have lots of other customers without due date rules, I suffer a lot with these 4 specific customers. If the invoice was created with the incorrect due date, the customer will reject it and ask for a new one, which means I need to cancel the invoice and submit a new one - sometimes near the payment date, moving the payment date to the next cycle/month and impacting my cash flow. Basically, 30% of the invoices I have in Kimai for these 4 customers are invoices that were cancelled.

Would this be possible or feasible to be implemented via a plugin?

**Comment by kevinpapst** [comment]
Hi, what a complicated mess.
I never heard of such strange rules, why don't you just force simple rules (e.g. 30 days from now onw) upon them?
Will they kick you out of the project if not? Usually large customers do not care about payment dates anyway. They have their payroll dates and use them, no matter what is on your invoice. 

Yes, your rules should be possible via plugin, but honestly: why?

**Reply by andrelam** [reply]
Yes, definitely a mess... But this is somewhat common in Brazil for companies that work with the government and also IT consulting companies. These payment terms are present in their contracts, and it's either a "accept it" or "don't work with us".

I'm not as experienced in PHP, but I'll take a look at the plug-in development examples.

Thanks!
